test/test_torch.py

```diff
@@ -978,6 +979,100 @@ def _test_norm(self, device):
     def test_norm(self):
         self._test_norm(self, device='cpu')
 
+    @staticmethod
+    def _test_nuclear_norm_axes(self, device='cpu'):
+        def check_single_nuclear_norm(x, axes):
+            if x.is_cuda and randrange(100) < 95:
+                return  # too many cpu <==> gpu copies
```

Comment:
What is the purpose of randrange?

If you are concerned about too many copies you can create a tensor on the CPU and pass it to this function, and then create a copy of the tensor on the device that you are testing for explicitly.

Something like this:
```python
x_device = x.to(device=device, copy=True) # x is a CPU tensor invariably.
```

Reply:
The tests are too slow, so some are skipped (they all pass here on my machine without randrange).

Reply:
So I think that the CUDA tests spend the majority of the time copying from GPU ==> CPU so that numpy arrays can be created and the results can be compared.

Reply:
Currently there are three copies when `x` is already on the GPU:

1) For creating the numpy array.

2) For copying the first result to CPU for `allclose()`.

3) For copying the second result to CPU for `allclose()`.

One can reduce it to two copies by copying `expected` to GPU, but it would probably not be much better.